From: srinivas pulukuru <srinivas.pulukuru@ti.com>
Date: Thu, 14 Apr 2011 12:13:43 +0000 (-0500)
Subject: OMAP4 DSS HDMI: Read EDID only on HPD high
X-Git-Tag: android-2.6.35-2.3-omap4.13.1~37
X-Git-Url: http://git.omapzoom.org/?p=kernel%2Fomap.git;a=commitdiff_plain;h=57c499d807ab7221e90fd7134df1bff700ed7325

OMAP4 DSS HDMI: Read EDID only on HPD high

1. No EDID read should happen on Physical connect as per the HDMI spec
Only on HPD high the EDID data is available and should only be read
at that time.
2. When custom timing is set EDID should only be read on HPD high.

Change-Id: I765c45c803f3372a267b300300f26f07567f71a8
Signed-off-by: srinivas pulukuru <srinivas.pulukuru@ti.com>
Signed-off-by: fabrice olivero <f-olivero@ti.com>
---

diff --git a/drivers/video/omap2/dss/hdmi.c b/drivers/video/omap2/dss/hdmi.c
index 49d75ca..73f6831 100644
--- a/drivers/video/omap2/dss/hdmi.c
+++ b/drivers/video/omap2/dss/hdmi.c
@@ -1440,6 +1440,8 @@ static void hdmi_work_queue(struct work_struct *ws)
 		DSSINFO("Physical Connect\n");
 
 		/* turn on clocks on connect */
+		edid_set = false;
+		custom_set = true;
 		hdmi_reconfigure(dssdev);
 		mutex_unlock(&hdmi.lock_aux);
 		hdmi_notify_pwrchange(HDMI_EVENT_POWERON);
@@ -1450,6 +1452,7 @@ static void hdmi_work_queue(struct work_struct *ws)
 		if (dssdev->state != OMAP_DSS_DISPLAY_ACTIVE)
 			/* HDMI is disabled, no need to process */
 			goto done;
+		custom_set = false;
 	}
 
 done:
@@ -2257,6 +2260,7 @@ static int hdmi_read_edid(struct omap_video_timings *dp)
 		ret = HDMI_CORE_DDC_READEDID(HDMI_CORE_SYS, edid,
 							HDMI_EDID_MAX_LENGTH);
 	if (ret != 0) {
+		edid_set = false;
 		printk("HDMI failed to read E-EDID\n");
 	} else {
 		if (!memcmp(edid, header, sizeof(header))) {
